holopy/scattering/imageformation.py

```diff
@@ -0,0 +1,233 @@
+
+    def _calculate_single_color_scattered_field(self, scatterer, schema):
+        if self.scattering_theory.can_handle(scatterer):
+            field = self._get_field_from(scatterer, schema)
+        elif isinstance(scatterer, Scatterers):
+            field = self._calculate_scattered_field_from_superposition(
+                scatterer.get_component_list(), schema)
+        else:
+            raise TheoryNotCompatibleError(self.scattering_theory, scatterer)
+        return self._pack_field_into_xarray(field, schema)
+
+    def _get_field_from(self, scatterer, schema):
+        """
+        Parameters
+        ----------
+        scatterer
+        schema : xarray
+            (it's always passed in as an xarray)
+
+        Returns
+        -------
+        raveled fields, shape (npoints = nx*ny = schema.shape.prod(), 3)
+        """
+        wavevector = get_wavevec_from(schema)
+        positions = self._transform_to_desired_coordinates(
+            schema, scatterer.center, wavevec=wavevector)
+        scattered_field = np.transpose(
+            self.scattering_theory.raw_fields(
+                positions,
+                scatterer,
+                medium_wavevec=wavevector,
+                medium_index=schema.medium_index,
+                illum_polarization=schema.illum_polarization)
+            )
+        phase = np.exp(-1j * wavevector * scatterer.center[2])
+        scattered_field *= phase
+        return scattered_field
+
+    def _pack_field_into_xarray(self, scattered_field, schema):
+        """Packs the numpy.ndarray, shape (N, 3) ``scattered_field`` into
+        an xr.DataArray, shape (N, 3). This function needs to pack the
+        fields [flat or point, vector], with the coordinates the
+        same as that of the schema."""
+        flattened_schema = flat(schema)  # now either point or flat
+        point_or_flat = self._is_detector_view_point_or_flat(flattened_schema)
+        coords = {
+            key: (point_or_flat, val.values)
+            for key, val in flattened_schema[point_or_flat].coords.items()}
+
+        coords.update(
+            {point_or_flat: flattened_schema[point_or_flat],
+             vector: ['x', 'y', 'z']})
+        scattered_field = xr.DataArray(
+            scattered_field, dims=[point_or_flat, vector], coords=coords,
+            attrs=schema.attrs)
+        return scattered_field
+
+    def _pack_scattering_matrix_into_xarray(
+            self, scat_matrs, r_theta_phi, schema):
+        flattened_schema = flat(schema)
+        point_or_flat = self._is_detector_view_point_or_flat(flattened_schema)
+        dims = [point_or_flat, 'Epar', 'Eperp']
+
+        coords = {point_or_flat: flattened_schema.coords[point_or_flat]}
+        coords.update({
+            'r': (point_or_flat, r_theta_phi[0]),
+            'theta': (point_or_flat, r_theta_phi[1]),
+            'phi': (point_or_flat, r_theta_phi[2]),
+            'Epar': ['S2', 'S3'],
+            'Eperp': ['S4', 'S1'],
```

Comment:
Yeah I have no idea what the thought process here was. This is just legacy code that I kept. 

I will say though that the output is user-facing (if `holopy.interface.calc_scat_matrix` is called), so I am hesitant to change the output.

Reply:
I don't have a problem breaking user-facing code if what was there before was flat out wrong. Anyone using this functionality was getting results that didn't make sense, at least in terms of naming. If we now give them an error I think that's actually a good thing because it makes them look more carefully at what's changed and what they should be doing.

Reply:
Sorry to post on the closed PR @briandleahy, but it looks like this hasn't been fixed in #379. If it's just the naming that is incorrect it's not a big deal, but I'm worried the incorrect naming is a sign that whoever wrote this code (possibly me) didn't understand what was going on when they wrote it and we should take a closer look. 

It's fine if you don't want to get into this now, but in that case we should raise an issue.

Reply:
Sorry, I thought I had already replied to this.

I'd rather change this later, because (i) I think this is separate issue than the refactor, and I'd like to keep the refactor atomic, and (ii) I have no idea what the design choice or use cases are for this bit of code, and I'm hesitant to change something without an understanding of what the code is supposed to be doing.

I'll raise an issue now.